### PR TITLE
Support for Anthropic handler

### PR DESCRIPTION
**Pull request by bala-ceg**
## Description

Anthropic LLM for MindsDB

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📄 This change requires a documentation update

### What is the solution?
Integrated Anthropic LLM for MindsDB as per ML handler framework 

### Working Screenshots

Screenshot 2023-07-17 at 3 56 22 PM

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation or created issues to update them.
- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

File: mindsdb/integrations/handlers/anthropic_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Anthropic Handler
+Anthropic ML handler integrates Anthropic LLMs with MindsDB. You can train your existing text data with Anthropic LLM
```
Comment (paxcema) [inline mindsdb/integrations/handlers/anthropic_handler/README.md:2]: > You can train your existing text data with Anthropic LLM

This doesn't seem accurate. This handler does not support fine-tuning (i.e. no `update()` method), so I would suggest rewording to something like:

> The Anthropic ML handler integrates Anthropic LLMs with MindsDB. You can use it to generate text completions with the popular Claude LLM family for your existing text data.

File: mindsdb/integrations/handlers/anthropic_handler/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# Anthropic
+Anthropic is a AI assitant framework for your usecases. In this handler, we use the Anthropic package, which is available in Python. More information about this Python client can be found (here)[https://github.com/anthropics/anthropic-sdk-python].
```
Comment (paxcema) [inline mindsdb/integrations/handlers/anthropic_handler/README.md:5]: Please correct this paragraph, there are a number of imprecisions here:

* Anthropic is an AI safety and research company. Claude is their assistant model.
* The package used is the Anthropic Python SDK.
* A bunch of typos, like "assitant" or "usecases". I recommend using something like Grammarly to detect these for an easy fix.

File: mindsdb/integrations/handlers/anthropic_handler/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+
+# Example Usage
+~~~ sql
+CREATE MODEL mindsdb.anthropic_test
```
Comment (paxcema) [inline mindsdb/integrations/handlers/anthropic_handler/README.md:10]: Please add the prerequisite `CREATE ML_ENGINE` command here.

File: mindsdb/integrations/handlers/anthropic_handler/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  api_key = 'your_api_key',
+  max_tokens = '300',
+  model = 'claude-2'
+~~~
+
+~~~ sql
+SELECT question, answer
+FROM mindsdb.anthropic_test
+WHERE question = 'How to solve the climate change crisis?';
+~~~
```
Comment (paxcema) [inline mindsdb/integrations/handlers/anthropic_handler/README.md:24]: Please add an example that uses the `JOIN` syntax. You can check other examples in the documentation for some public DBs and tables we already expose with text datasets.

File: mindsdb/integrations/handlers/anthropic_handler/anthropic_handler.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+class AnthropicHandler(BaseMLEngine):
+    """
+    Integration with the Anthropic LLM Python Library
+    """
+    name = 'anthropic'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.default_chat_model = 'claude-2'
+        self.supported_chat_model = ['claude-1', 'claude-2']
```
Comment (paxcema) [inline mindsdb/integrations/handlers/anthropic_handler/anthropic_handler.py:24]: Nitpick, but I recommend renaming to `supported_chat_models`, plural.

File: mindsdb/integrations/handlers/anthropic_handler/anthropic_handler.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        """ 
+        connects with anthropic api to predict the answer for the particular question
+
+        """ 
+
+        args = self.model_storage.json_get('args')
+
+        api_key = self._get_anthropic_api_key(args)
+
+        anthropic = Anthropic(api_key=api_key,)
```
Comment (paxcema) [inline mindsdb/integrations/handlers/anthropic_handler/anthropic_handler.py:105]: I'm not sure how slow this connection may be, but it wouldn't hurt to move it back to `predict()` as e.g. `self.connection`, and thus avoid recreating it for every row.

File: mindsdb/integrations/handlers/anthropic_handler/README.md (hunk 2 above)
Comment (paxcema) [reply mindsdb/integrations/handlers/anthropic_handler/README.md:5]: I would additionally put a disclaimer in bold for users to first ensure they've read the evaluation terms that Anthropic defines [here](https://docs.anthropic.com/claude/reference/getting-started-with-the-api#evaluation--going-live-with-the-api).

**Review by paxcema — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/anthropic_handler/README.md (hunk 1 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/anthropic_handler/README.md:2]: Thanks for the review, I have made the necessary changes

File: mindsdb/integrations/handlers/anthropic_handler/README.md (hunk 2 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/anthropic_handler/README.md:5]: Thanks for the review, I have made the necessary changes

File: mindsdb/integrations/handlers/anthropic_handler/README.md (hunk 3 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/anthropic_handler/README.md:10]: Thanks for the review, I have made the necessary changes

File: mindsdb/integrations/handlers/anthropic_handler/README.md (hunk 4 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/anthropic_handler/README.md:24]: Thanks for the review, I have made the necessary changes

File: mindsdb/integrations/handlers/anthropic_handler/anthropic_handler.py (hunk 5 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/anthropic_handler/anthropic_handler.py:24]: Thanks for the review, I have made the necessary changes

File: mindsdb/integrations/handlers/anthropic_handler/anthropic_handler.py (hunk 6 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/anthropic_handler/anthropic_handler.py:105]: Thanks for the review, I have made the necessary changes

**Comment by bala-ceg** [comment]
Screenshot 2023-07-19 at 10 01 27 AM

File: mindsdb/integrations/handlers/anthropic_handler/README.md [hunk 7]
```diff
@@ -1,11 +1,23 @@
 # Anthropic Handler
-Anthropic ML handler integrates Anthropic LLMs with MindsDB. You can train your existing text data with Anthropic LLM
+The Anthropic ML handler integrates Anthropic LLMs with MindsDB. You can use it to generate text completions with the popular Claude LLM family for your existing text data.
 
 # Anthropic
-Anthropic is a AI assitant framework for your usecases. In this handler, we use the Anthropic package, which is available in Python. More information about this Python client can be found (here)[https://github.com/anthropics/anthropic-sdk-python].
+Anthropic is an AI safety and research company. Claude is their assistant model.The package used is the Anthropic Python SDK.More information about this Python client can be found (here)[https://github.com/anthropics/anthropic-sdk-python].
+
+*Note:* For more information on read the (here) [https://docs.anthropic.com/claude/reference/getting-started-with-the-api#evaluation--going-live-with-the-api]
```
Comment (paxcema) [inline mindsdb/integrations/handlers/anthropic_handler/README.md]: The syntax on both links is backwards. Markdown supports links with hyperlinked text in square brackets and the link in round brackets, [like so](https://mindsdb.com).

**Review by paxcema — APPROVED** [review]
Small issue with link formatting on .md files, but apart from that this looks good to merge 👍, thanks for your contribution @bala-ceg! Let me know once you've fixed the links and I'll merge.

**Comment by bala-ceg** [comment]
Thanks, @paxcema for reviewing. I fixed the link formatting issues on `README.md`

**Comment by paxcema** [comment]
Thanks for your contribution @bala-ceg! Merging now.
